nucypher/characters/lawful.py

```diff
@@ -527,7 +527,8 @@ def generate_work_orders(self, map_id, *capsules, num_ursulas=None):
                     arrangement_id, capsules_to_include, ursula, self)
                 generated_work_orders[node_id] = work_order
                 # TODO: Fix this. It's always taking the last capsule
-                self._saved_work_orders[node_id][capsule] = work_order
+                if cache:
+                    self._saved_work_orders[node_id][capsule] = work_order
```

Comment:
Is this a possible unhanded `KeyError`?

Reply:
I don't immediately see how - what's your thinking?

Reply:
Nothing specific, just a general curiosity whenever I see this kind of pattern.  In this case - it looks like there is only setter behavior, so looking good I think.